web/ui_utils.js

```diff
@@ -355,22 +355,25 @@ function noContextMenuHandler(e) {
 
 /**
  * Returns the filename or guessed filename from the url (see issue 3455).
- * url {String} The original PDF location.
- * defaultFilename {string} The value to return if the file name is unknown.
- * @return {String} Guessed PDF file name.
+ * @param {string} url - The original PDF location.
+ * @param {string} defaultFilename - The value returned if the filename is
+ *   unknown, or the protocol is unsupported.
+ * @returns {string} Guessed PDF filename.
  */
-function getPDFFileNameFromURL(url, defaultFilename) {
-  if (typeof defaultFilename === 'undefined') {
-    defaultFilename = 'document.pdf';
+function getPDFFileNameFromURL(url, defaultFilename = 'document.pdf') {
+  if (url.indexOf('data:') === 0) {
```

Comment:
Per https://tools.ietf.org/html/rfc3986#section-3.1 , schemes are case-insensitive. I recommend to create the isDataSchema(url) function above and also skip allowed whitespaces, something like:
```
var i = 0; while (i < url.length && url[i].trim() == '') i++;
return url.substr(i, 5).toLowerCase() === 'data:';
```